File: app/src/main/java/com/song/deviceinfo/ui/about/AboutViewModel.java
package com.song.deviceinfo.ui.about;

import android.content.Context;

import com.song.deviceinfo.R;
import com.song.deviceinfo.utils.Constants;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class AboutViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public AboutViewModel() {
        mText = new MutableLiveData<>();
    }

    public LiveData<String> getText() {
        return mText;
    }

    public void setData(Context context) {
        mText.setValue(String.format(context.getString(R.string.about_version), Constants.VERSION));
    }
}

File: app/src/main/java/com/song/deviceinfo/ui/about/AboutViewModel.kt
package com.song.deviceinfo.ui.about

import android.content.Context
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.song.deviceinfo.R
import com.song.deviceinfo.utils.Constants

class AboutViewModel : ViewModel() {
    private val mText = MutableLiveData<String?>()
    
    val text: LiveData<String?>
        get() = mText
    
    fun setData(context: Context) {
        mText.value = String.format(context.getString(R.string.about_version), Constants.VERSION)
    }
}
